tests/test_nightReport.py

```diff
@@ -0,0 +1,234 @@
+        sortedKeys = sorted(list(sortedData.keys()))
+        self.assertEqual(sortedKeys, list(self.report.data.keys()))
+        return
+
+    def test_getExpRecordDictForDayObs(self):
+        """Test getExpRecordDictForDayObs.
+
+        Test it returns a dict of dicts, keyed by integer seqNums.
+        """
+        expRecDict = self.report.getExpRecordDictForDayObs(self.dayObs)
+        self.assertIsInstance(expRecDict, dict)
+        self.assertGreaterEqual(len(expRecDict), 1)
+
+        # check all the keys are ints
+        seqNums = list(expRecDict.keys())
+        self.assertTrue(all(isinstance(s, int) for s in seqNums))
+
+        # check all the values are dicts
+        self.assertTrue(all(isinstance(expRecDict[s], dict) for s in seqNums))
+        return
+
+    def test_getObsInfoAndMetadataForSeqNum(self):
+        """Test that getObsInfoAndMetadataForSeqNum returns the correct types.
+        """
+        seqNum = self.seqNums[0]
+        obsInfo, md = self.report.getObsInfoAndMetadataForSeqNum(seqNum)
+        self.assertIsInstance(obsInfo, ObservationInfo)
+        self.assertIsInstance(md, dict)
+        return
+
+    def test_rebuild(self):
+        """Test that rebuild does nothing, as no data will be being added.
+
+        NB Do not call full=True on this, as it will double the length of the
+        tests and they're already extremely slow.
+        """
+        lenBefore = len(self.report.data)
+        self.report.rebuild()
+        self.assertEqual(len(self.report.data), lenBefore)
+        return
+
+    def test_getExposureMidpoint(self):
+        """Test the exposure midpoint calculation
+        """
+        midPoint = self.report.getExposureMidpoint(self.seqNums[0])
+        record = self.report.data[self.seqNums[0]]
+        self.assertGreater(midPoint, record['datetime_begin'].mjd)
+        self.assertLess(midPoint, record['datetime_end'].mjd)
+        return
+
+    def test_getTimeDeltas(self):
+        """Test the time delta calculation returns a dict.
+        """
+        dts = self.report.getTimeDeltas()
+        self.assertIsInstance(dts, dict)
+        return
+
+    def test_makeStarColorAndMarkerMap(self):
+        """Test the color map maker returns a dict of ColorAndMarker objects.
+        """
+        cMap = self.report.makeStarColorAndMarkerMap(self.report.stars)
+        self.assertEqual(len(cMap), len(self.report.stars))
+        self.assertIsInstance(cMap, dict)
+        values = list(cMap.values())
+        self.assertTrue(all(isinstance(value, ColorAndMarker) for value in values))
+        return
+
+    def test_printObsTable(self):
+        """Test that a the printObsTable() method prints out the correct
+        number of lines.
+        """
+        with mock.patch('sys.stdout') as fake_stdout:
+            self.report.printObsTable()
+
+        # newline for each row plus header line, plus the line with dashes
+        self.assertEqual(len(fake_stdout.mock_calls), 2*(self.nImages+2))
```

Comment:
+ with no spaces.

Reply:
The `+` is the problem here without the spaces, not the `*`, right?